### ISSUE TITLE
Proxy bypass not working in 1.15.0

### ISSUE THREAD

**Issue by cletusw** [webpack/webpack-dev-server#560]
A `webpack.config.js` like this (mostly pulled from [the docs](http://webpack.github.io/docs/webpack-dev-server.html))

```javascript
var path = require("path");
module.exports = {
  entry: {
    app: ["./app/main.js"]
  },
  output: {
    path: path.resolve(__dirname, "build"),
    publicPath: "/assets/",
    filename: "bundle.js"
  },
  devServer: {
    proxy: {
      '/': {
        bypass: function(){
          return '/test.html';
        }
      }
    }
  }
};
```

works fine with `webpack-dev-server@1.14.1`, but after updating to `1.15.0`, I get

```
$ node_modules/.bin/webpack-dev-server --content-base build/
/Users/claytonwatts/dev/test/node_modules/webpack-dev-server/node_modules/http-proxy-middleware/lib/config-factory.js:42
        throw new Error('[HPM] Missing "target" option. Example: {target: "http://www.example.org"}');
        ^

Error: [HPM] Missing "target" option. Example: {target: "http://www.example.org"}
    at Object.createConfig (/Users/claytonwatts/dev/test/node_modules/webpack-dev-server/node_modules/http-proxy-middleware/lib/config-factory.js:42:15)
    at new HttpProxyMiddleware (/Users/claytonwatts/dev/test/node_modules/webpack-dev-server/node_modules/http-proxy-middleware/lib/index.js:16:45)
    at module.exports (/Users/claytonwatts/dev/test/node_modules/webpack-dev-server/node_modules/http-proxy-middleware/index.js:4:12)
    at /Users/claytonwatts/dev/test/node_modules/webpack-dev-server/lib/Server.js:184:9
    at Array.forEach (native)
    at Object.Server.features.proxy (/Users/claytonwatts/dev/test/node_modules/webpack-dev-server/lib/Server.js:171:19)
    at Server.<anonymous> (/Users/claytonwatts/dev/test/node_modules/webpack-dev-server/lib/Server.js:274:20)
    at Array.forEach (native)
    at new Server (/Users/claytonwatts/dev/test/node_modules/webpack-dev-server/lib/Server.js:273:40)
    at Object.<anonymous> (/Users/claytonwatts/dev/test/node_modules/webpack-dev-server/bin/webpack-dev-server.js:183:1)
```

Looks like [this](https://github.com/webpack/webpack-dev-server/commit/efd32a05944499b0c8165da6ff75519fb8fe21a6) is the commit that adds `http-proxy-middleware`.

**Comment by SpaceK33z** [comment]
Confirmed this is also happening with `2.1.0-beta.0`.

**Comment by SpaceK33z** [comment]
Could you try out if #563 works for you?

**Comment by cletusw** [comment]
@SpaceK33z I'm now getting `Fatal error: ENOENT: no such file or directory, open '<project>/node_modules/webpack-dev-server/client/live.bundle.js'`

Not sure if that's from this or if I'm missing something else. I'll look a little deeper.

**Comment by SpaceK33z** [comment]
@cletusw, you need to run `npm i && npm run prepublish` in `node_modules/webpack-dev-server`. You could also copy/paste the contents of `lib/Server.js` in my PR (hacky hacky ;p).

**Comment by cletusw** [comment]
@SpaceK33z That did it! Working fine now, thanks!

**Comment by SpaceK33z** [comment]
@cletusw, awesome! There are a few other small bugs in `1.15.0` that I want to fix first, but a release should happen soon.

**Comment by seansfkelley** [comment]
Would it be possible to make a release now, just for this, since it's a P0? A lot of my team members keep hitting this issue and wasting time debugging until they find this ticket.

**Comment by txangel** [comment]
@SpaceK33z we had to fix out version to 1.14.1 (instead of ^1.14.1) for now, which is not a very good practice. 
Do you think it would be possible to get a release sometime this week?

On top of that you might want to increase the test coverage here to reduce the risk of having another breaking change when upgrading the minor version. 
Plenty of teams depend on your code mate; the price of fame (:

**Comment by SpaceK33z** [comment]
@txangel, I'm trying, but it is out of my hands (I have no permissions to publish to npm).

You're right in that there should be test coverage. It is very hard to test the dev-server thoroughly though, but there are some things we could test in isolation.

**Comment by SpaceK33z** [comment]
This has been released as part of  `1.15.1`.

**Comment by seansfkelley** [comment]
Thank you!
